akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -220,4 +220,11 @@ object Sink extends SinkApply {
     new Sink(new AcknowledgeSink(bufferSize, DefaultAttributes.acknowledgeSink, shape("AcknowledgeSink"), timeout))
   }
 
+  /**
+   * Crates a `Sink` that is materialized into `Source` to connect with another flow.
+   * This sink gets backpressure from second flow and does not have internal buffer.
+   */
+  def pipedSink[T](): Sink[T, Source[T, Unit]] =
+    Sink.publisher[T].mapMaterializedValue(Source(_)).named("PipedSink")
```

Comment:
> Why does the upper part of the stream have to be running after materialization of the sink?

Because materialization is about running. It is even called `run()`. If it would not start the upper part of the stream that would mean the introduction of *another* lifted representation layer, and I am already slightly annoyed with having one.

> Why can't materialization say "oh, I don't actually have a sink yet, so there's no point in starting the source yet, so I won't materialize it"? I don't see why it needs to run the upstream.

Because the materializer only deals with Publishers and Subscribers at that point and changing this would lead to unnecessary complications and serious performance implications, given that it would result in double/triple indirections in various places.

The streams philosophy (as the API is currently) favors encoding dependencies in your modules directly, rather than in function arguments. I.e. if your code needs a Source, then your code should not accept a Source as a parameter to a function, but return a Sink instead. While this ideal cannot be met in 100% of the cases, in general the unit of modularity is an Akka Streams component, not a function. Also, this prefers mostly bottom-up organization (although top down is also possible).